client/a8c-for-agencies/hooks/use-track-provisioning-sites.ts

```diff
@@ -0,0 +1,63 @@
+import { useState, useEffect, useCallback } from 'react';
+
+const TTL_DURATION = 300000; // 5 minutes in miliseconds
+
+type ProvisioningSite = {
+	id: number;
+	migration?: boolean;
+	ttl: number;
+};
+
+export default function useTrackProvisioningSites() {
+	const [ provisioningSites, setProvisioningSites ] = useState< ProvisioningSite[] >( [] );
+
+	const reload = useCallback( () => {
+		const storedSites = localStorage.getItem( 'provisioningSites' );
+		if ( storedSites ) {
+			const sites: ProvisioningSite[] = JSON.parse( storedSites );
+			// We remove the site from tracking if it is more than 5 minutes.
+			const activeSites = sites.filter( ( { ttl } ) => ttl >= new Date().getTime() );
+
+			setProvisioningSites( activeSites );
+			localStorage.setItem( 'provisioningSites', JSON.stringify( activeSites ) );
+		}
+	}, [] );
+
+	const trackSiteId = useCallback(
+		( siteId: number, migration?: boolean ) => {
+			const updatedSites = [
+				...provisioningSites,
+				{ id: siteId, migration, ttl: new Date().getTime() + TTL_DURATION },
+			];
+
+			localStorage.setItem( 'provisioningSites', JSON.stringify( updatedSites ) );
+			window.dispatchEvent( new Event( 'storage' ) );
+		},
+		[ provisioningSites ]
+	);
+
+	const untrackSiteId = useCallback(
+		( siteId: number ) => {
+			const updatedSites = provisioningSites.filter( ( { id } ) => id !== siteId );
+
+			localStorage.setItem( 'provisioningSites', JSON.stringify( updatedSites ) );
+			window.dispatchEvent( new Event( 'storage' ) );
```

Comment:
We will use Windows Event Dispatch to ensure that the list is updated when the storage object is changed. This is crucial to notify all consuming components of the change.